### PR TITLE
Add Text support

### PR DESCRIPTION
**Pull request by jeffin143**
- [x] Add support for logging single string value

### CODE REVIEW

**Comment by toshal-a** [comment]
I think this PR also needs a rebase. Also add unit test for this PR.

File: examples/text.md
Comment (toshal-a) [inline examples/text.md]:
```suggestion
The output would be similar to: 
```

**Review by toshal-a — COMMENTED** [review]
This needs a rebase. Left one small suggestion.

**Review by birm — APPROVED** [review]
I think it could use a test case, but it looks good.

File: examples/text.md
Comment (toshal-a) [inline examples/text.md]:
```suggestion
These examples help you to understand SummaryWriter::Text() API in depth.
```

File: examples/text.md
Comment (toshal-a) [inline examples/text.md]: We can avoid the global loading of these namespaces.

File: include/filewriter/summarywriter_impl.hpp
Comment (toshal-a) [inline include/filewriter/summarywriter_impl.hpp]: I think you should use two spaces (one tab) instead of four spaces. Indentation error.

File: examples/text.md
Comment (toshal-a) [inline examples/text.md]: Same over here I know this is just an example so optional. But In this also four spaces are there.

**Review by toshal-a — COMMENTED** [review]
Hi @jeffin143, I think @birm is right. It would be great if we add a unit test for this support. Test  similar to the example will also be fine.

File: tests/filewriter_test.cpp
Comment (toshal-a) [inline tests/filewriter_test.cpp]: Indentation error.

File: tests/filewriter_test.cpp
Comment (toshal-a) [inline tests/filewriter_test.cpp]: Same over here indentation problem

File: tests/filewriter_test.cpp
Comment (toshal-a) [inline tests/filewriter_test.cpp]: I think the function call should be two tabs ahead only

File: tests/filewriter_test.cpp
Comment (toshal-a) [inline tests/filewriter_test.cpp]: Same on this function calls.

**Review by toshal-a — COMMENTED** [review]
Left some comments.

File: examples/text.md
Comment (toshal-a) [inline examples/text.md]:
```suggestion
                                     mlboard::Filewriter& fw)
```

File: include/filewriter/filewriter_impl.hpp
Comment (toshal-a) [inline include/filewriter/filewriter_impl.hpp]: Indentation mismatch.

File: include/filewriter/summarywriter.hpp
Comment (toshal-a) [inline include/filewriter/summarywriter.hpp:38]:
```suggestion

  /**
```

File: examples/text.md
Comment (toshal-a) [inline examples/text.md]:
```suggestion
The API accepts `tag`, `step`, `value` (String value) and `mlboard::Filewriter` object.
```

File: tests/filewriter_test.cpp
Comment (toshal-a) [inline tests/filewriter_test.cpp]:
```suggestion
  REQUIRE(f1.FileName().substr(8,f1.FileName().length()) ==
      f2.FileName().substr(8,f2.FileName().length()));

  // Remove event files.
```

File: tests/filewriter_test.cpp
Comment (toshal-a) [inline tests/filewriter_test.cpp]:
```suggestion
    mkdir("_temp1_", 0777);
    mkdir("_temp2_", 0777);
```

File: tests/filewriter_test.cpp
Comment (toshal-a) [inline tests/filewriter_test.cpp]:
```suggestion
  REQUIRE(f1.FileName().substr(8, f1.FileName().length()) ==
      f2.FileName().substr(8, f2.FileName().length()));
```

File: tests/main.cpp
Comment (toshal-a) [inline tests/main.cpp]:
```suggestion
      mkdir("_templogs", 0777);
```

File: tests/summarywriter_test.cpp
Comment (toshal-a) [inline tests/summarywriter_test.cpp]: Indentation mismatch.

**Review by toshal-a — COMMENTED** [review]
Hi, left some comments. Let's merge this after #6 gets merged.

File: include/filewriter/summarywriter_impl.hpp
Comment (toshal-a) [inline include/filewriter/summarywriter_impl.hpp:53]:
```suggestion

template<typename Filewriter>
```

File: tests/filewriter_test.cpp
Comment (toshal-a) [inline tests/filewriter_test.cpp]:
```suggestion
 * @file tests/filewriter_test.cpp
```

File: tests/summarywriter_test.cpp
Comment (toshal-a) [inline tests/summarywriter_test.cpp]: Let's not use add this as text.
```suggestion
      "Test case for checking text support in mlboard.", *f1);
```

File: tests/main.cpp
Comment (toshal-a) [inline tests/main.cpp]: Two things it is duplicate code. Might be got introduced while rebasing. 

The next thing is that in my os `_templogs` directory was not getting deleted. The reason is it's created in `main.cpp` so the directory contains some temp files which need to go away. When I tried creating the directory in the `summary_writer_test.cpp` the issue gets resolved i.e the directory gets deleted correctly. Now the challenge is how to handle it in `summary_writer_test.cpp` file for that what we can do is to create it in the constructor of SummaryWriterTestFixture
The code would get slightly duplicated but I think we might handle it by using a static flag. Let me know what you think.

The code tried is like this
```
class SummaryWriterTestsFixture {
 
  public:
  SummaryWriterTestsFixture() {
    #if defined(_WIN32)
        _mkdir("_templogs");
    #else
        mkdir("_templogs", 0777);
    #endif
  };

 
  static mlboard::FileWriter* f1;
  static size_t currentSize;
 };

```

File: tests/summarywriter_test.cpp
Comment (toshal-a) [inline tests/summarywriter_test.cpp]:
```suggestion
 * @file tests/summarywriter_test.cpp
```

File: tests/summarywriter_test.cpp
Comment (toshal-a) [inline tests/summarywriter_test.cpp]:
```suggestion
class SummaryWriterTestsFixture 
{
```

File: tests/summarywriter_test.cpp
Comment (toshal-a) [inline tests/summarywriter_test.cpp]:
```suggestion
};
```

File: examples/faq.md
Comment (toshal-a) [inline examples/faq.md]:
```suggestion
The above code would probably throw error `No event timestamp could be found`, and hence to avoid such error please avoid reassigning the `Filewriter` object.
```

File: examples/faq.md
Comment (toshal-a) [inline examples/faq.md]:
```suggestion
- There could be chances that there are race condition where the main thread is exiting before the writer thread, hence you could avoid this using `.close()` method of `mlboard::Filewriter` object.
```

File: examples/faq.md
Comment (toshal-a) [inline examples/faq.md]: These lines are not comming properly indented I think checkout display rich diff option to see what I am trying to say.

File: examples/faq.md
Comment (toshal-a) [inline examples/faq.md]: Same about these lines.

File: examples/faq.md
Comment (toshal-a) [inline examples/faq.md]:
```suggestion
3. Error: Unable to get first event timestamp for run:
```

File: examples/faq.md
Comment (toshal-a) [inline examples/faq.md]:
```suggestion
- If you get the following error:
```

File: examples/faq.md
Comment (toshal-a) [inline examples/faq.md]:
```suggestion
- If you get the following error:
```

File: examples/faq.md
Comment (toshal-a) [inline examples/faq.md]: May be it's better if we highlight these headings. Let me know what you think.

**Review by toshal-a — COMMENTED** [review]
Hey @jeffin143, the changes you have made look good. Good to see the faq file but I think I never encountered these errors. I think, this would be the final review from my side. :+1:

File: examples/faq.md
Comment (jeffin143) [reply examples/faq.md]: Can you help me this, I am not sure what is the error

Comment (toshal-a) [reply examples/faq.md]: https://meta.stackexchange.com/questions/348746/were-switching-to-commonmark

Search for `The most noticeable changes will be around lists, nested lists, headlines and blockquotes.` in the above doc you will get the solution.

File: tests/main.cpp
Comment (toshal-a) [reply tests/main.cpp]: Hey @jeffin143, Looks like you missed to make this change. Please make this change. The PR is ready to get merged.

Comment (jeffin143) [reply tests/main.cpp]: Shouldn't it be destructor Since constructor would only run when the object is created

Comment (jeffin143) [reply tests/main.cpp]: Oh I understood

**Review by toshal-a — APPROVED** [review]

File: tests/filewriter_test.cpp
Comment (birm) [inline tests/filewriter_test.cpp]: Why did you remove "the" here?

**Review by birm — APPROVED** [review]
One nitpick, but otherwise great work!

File: tests/filewriter_test.cpp
Comment (toshal-a) [inline tests/filewriter_test.cpp]:
```suggestion
  // Both files created at the same time and hence there should be only one file.
```

**Review by toshal-a — COMMENTED** [review]
Okay I am commiting the small change and merging this in.

**Pull request merged**
